projects/gateway/pkg/syncer/translator_syncer.go

```diff
@@ -402,6 +408,10 @@ func (s *statusSyncer) extractCurrentReports() (reporter.ResourceReports, map[re
 func (s *statusSyncer) syncStatus(ctx context.Context) error {
 	allReports, inputResourceBySubresourceStatuses, localInputResourceLastStatus := s.extractCurrentReports()
 
+	// reset the resource metrics before we write the new statuses, this ensures that
```

Comment:
cant this cause flickering to an operator?
Why cant we set new statuses and then clear all out of compliance ones after?

Reply:
We are not able to clear out metrics at the label/attribute level. Only at the metric-level (view), which includes all labels).